File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/DBG-1/kotlin/dbg1.kt
// DBG-1
class Item(val count: Int, val name: String)

const val ITERATIONS: Long = 300000000L
const val MODULUS: Long = 1000000007L

fun accumulate(items: ArrayList<Item>, iterations: Long): Long {
    var total: Long = 0
    var i: Long = 0
    while (i < iterations) {
        for (item in items) {
            total = (total * 31 + item.count + item.name.length) % MODULUS
        }
        i++
    }
    return total
}

fun main() {
    val items = ArrayList<Item>()
    items.add(Item(7, "alpha"))
    items.add(Item(11, "bravo"))
    items.add(Item(13, "charlie"))
    val checksum = accumulate(items, ITERATIONS)
    println("checksum=" + checksum)
}

File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/DBG-1/zig/dbg1.zig
// DBG-1
const std = @import("std");

const ITERATIONS: i64 = 300000000;
const MODULUS: i64 = 1000000007;

const Item = struct {
    count: i64,
    name: []const u8,
};

fn accumulate(items: []const Item, iterations: i64) i64 {
    var total: i64 = 0;
    var i: i64 = 0;
    while (i < iterations) : (i += 1) {
        for (items) |item| {
            total = @rem(total * 31 + item.count + @as(i64, @intCast(item.name.len)), MODULUS);
        }
    }
    return total;
}

pub fn main() !void {
    var gpa = std.heap.DebugAllocator(.{}){};
    const allocator = gpa.allocator();
    var items: std.ArrayList(Item) = .empty;
    defer items.deinit(allocator);
    try items.append(allocator, Item{ .count = 7, .name = "alpha" });
    try items.append(allocator, Item{ .count = 11, .name = "bravo" });
    try items.append(allocator, Item{ .count = 13, .name = "charlie" });
    const checksum = accumulate(items.items, ITERATIONS);
    std.debug.print("checksum={d}\n", .{checksum});
}
